python/datasources/acs_health_insurance.py

```diff
@@ -0,0 +1,657 @@
+
+
+def log_response(resp):
+    print("-"*50)
+    print(f'Get Request ({resp.status_code}): {resp.url}')
+    print("-"*50)
+
+
+class AcsHealhInsuranceIngestor:
+
+    # Initialize variables in class instance, also merge all metadata so that lookup of the
+    # prefix, suffix combos can return the entire metadata
+    def __init__(self, base):
+        self.baseUrl = base
+        self.data = {}
+        self.metadata = {}
+        self.buildMetadataList()
+
+    # Gets standardized filename
+    def getFilename(self, sex, race, isCounty):
+        if(race is not None):
+            return "HEALTH_INSURANCE_BY_RACE_{0}_{1}.json".format("STATE" if isCounty else "COUNTY", race)
+        else:
+            return "HEALTH_INSURANCE_BY_SEX_{0}_{1}.json".format("STATE" if isCounty else "COUNTY", sex)
+
+    # Method to output <Filename.csv>.  Used for debugging purposes.
+    def write_local_files_debug(self):
+        self.getStateFipsMap()
+        self.getCountyFipsMap()
+        self.getHealthInsuranceDataBySex()
+        self.getHealthInsuranceDataByRace()
+        self.splitDataFrames()
+
+        with open('total_health_insurance.json', 'w') as f:
+            print(json.dumps(
+                {str(k): v for k, v in self.data.items()}, indent=4), file=f)
+
+        for table_name, df in self.frames.items():
+            df.to_csv(table_name + ".csv", index=False)
+
+    # Uploads the ACS data to GCS by providing
+    # the ACS Base URL
+    # Acs Query Params
+    # Standardized Filename
+    #
+    # An example file created in GCS:
+    # HEALTH_INSURANCE_BY_RACE_COUNTY_WHITE_ALONE.json
+
+    def upload_to_gcs(self, bucket):
+        maleStateParams = format_params(
+            HEALTH_INSURANCE_BY_SEX_GROUPS_PREFIX, HEALTH_INSURANCE_BY_SEX_MALE_SUFFIXES)
+        fileDiff = url_file_to_gcs.url_file_to_gcs(
+            self.baseUrl, maleStateParams, bucket,
+            self.getFilename(Sex.MALE, None, False))
+
+        femaleStateParams = format_params(
+            HEALTH_INSURANCE_BY_SEX_GROUPS_PREFIX, HEALTH_INSURANCE_BY_SEX_FEMALE_SUFFIXES)
+        fileDiff = url_file_to_gcs.url_file_to_gcs(
+            self.baseUrl, femaleStateParams, bucket,
+            self.getFilename(Sex.FEMALE, None, False)) and fileDiff
+
+        maleCountyParams = format_params(
+            HEALTH_INSURANCE_BY_SEX_GROUPS_PREFIX, HEALTH_INSURANCE_BY_SEX_MALE_SUFFIXES, True)
+        fileDiff = url_file_to_gcs.url_file_to_gcs(
+            self.baseUrl, maleCountyParams, bucket,
+            self.getFilename(Sex.MALE, None, True)) and fileDiff
+
+        femaleCountyParams = format_params(
+            HEALTH_INSURANCE_BY_SEX_GROUPS_PREFIX, HEALTH_INSURANCE_BY_SEX_FEMALE_SUFFIXES, True)
+        fileDiff = url_file_to_gcs.url_file_to_gcs(
+            self.baseUrl, femaleCountyParams, bucket,
+            self.getFilename(Sex.FEMALE, None, True)) and fileDiff
+
+        for prefix in HEALTH_INSURANCE_BY_RACE_GROUP_PREFIXES:
```

Comment:
It would be helpful to add a comment above this block explaining what it does

Reply:
Done